students/cfarris/lessons/lesson07/assignment/test_parallel.py

```diff
@@ -0,0 +1,156 @@
+#!/usr/bin/env Python3
+
+'''
+Pytest for parallel.py
+'''
+
+# pylint: disable-msg=redefined-outer-name
+# pylint: disable-msg=invalid-name
+# pylint: disable-msg=too-many-locals
+
+from pymongo import MongoClient
+import pytest
+from parallel import (import_data,
+                      _read_csv,
+                      show_avail_products,
+                      show_rentals,
+                      check_redundancy)
+
+
+class MongoDBConnection():
+    """Connect to MongoDB"""
+
+    def __init__(self, host='127.0.0.1', port=27017):
+
+        self.host = host
+        self.port = port
+        self.connection = None
+
+    def __enter__(self):
+        self.connection = MongoClient(self.host, self.port)
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.connection.close()
```

Comment:
You know you don't have to copy and paste MongoDBConnection everywhere, just import it. :) In fact in this exercise, you have a lot of copied code. Get in the habit of using stuff you already have, instead of copying it. How much of this test file is different from the linear one? Could you just run it twice with different files?